www/app/login/login.controller.js

```diff
@@ -39,18 +40,26 @@
         password === userInfo.password;
     }
 
-    function authenticate(username, password) {
-
-      LoginFirebaseService.getUserInfo(username)
-        .then(function(userInfo) {
-          if (_credentialsAreCorrect(username, userInfo, password)) {
-            _authSuccess();
-            RevisionService.setUsername(username);
-          }else {
-            authError();
-          }
-        });
+    function authenticate(username, password){
+      if(areEmptyCredentials(username, password)){
+        LoginFirebaseService.logIn(username)
+            .then(function(user){
+              if(user.hasOwnProperty('username') && angular.equals(user.password, password)) {
+                LoginFirebaseService.setAuthUser(username);
+                _authSuccess();
+              } else {
+                _authError();
+              }
+            })
+            .catch(_authError)
+      }
     }
 
+    function areEmptyCredentials(username, password){
+      if(username != "" && password != ""){
+        return true;
+      }
+      return false;
+    }
```

Comment:
You could handle this in the view, by creating an `<form>` and set the `inputs` to be `required`. And finally in the `Ok button` you could do something like this:`ng-disabled="myForm.$invalid"`.